drivers/serial/Kconfig.apbuart

```diff
@@ -9,3 +9,8 @@ config UART_APBUART
 	select SERIAL_SUPPORT_INTERRUPT
 	help
 	  This option enables the APBUART driver for LEON processors.
+
+config UART_APBUART_NGULTRA
+	bool "Driver config for NG-Ultra"
```

Comment:
This feels a little hacky, the "scaler" value should be derived from the dt.

Reply:
Do you recommend to remove the UART_APBUART_NGULTRA config & define a specific property for the scaler?

Reply:
> Do you recommend to remove the UART_APBUART_NGULTRA config & define a specific property for the scaler?

yes, and ideally in devicetree -- something like:

https://github.com/zephyrproject-rtos/zephyr/blob/3322aad03b58a1120ed72ebae7ed3596ad4547cd/boards/arm/nucleo_f334r8/nucleo_f334r8.dts#L102

Reply:
My previous config is useless...

The current scaler formula from set_baud() function:
`scaler = (((core_clk_hz * 10) / (baud * 8)) - 5) / 10;`
is equivalent to:
`scaler = (core_clk_hz / (baud * 8)) - 0.5;`
which is different from get_baud() function:
`return core_clk_hz / ((scaler + 1) * 8);`
One of them must be false...

According to my datasheet, test on my boards & linux driver (end of the apbuart_set_termios function), the one from get_baud() function is true